cmd/image-builder/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"io"
+	"log"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/osbuild/images/pkg/arch"
+)
+
+var osStdout io.Writer = os.Stdout
+
+func cmdListImages(cmd *cobra.Command, args []string) error {
+	filter, err := cmd.Flags().GetStringArray("filter")
+	if err != nil {
+		return err
+	}
+	format, err := cmd.Flags().GetString("format")
+	if err != nil {
+		return err
+	}
+
+	return listImages(osStdout, format, filter)
+}
+
+func cmdManifest(cmd *cobra.Command, args []string) error {
+	distroName := args[0]
+	imgType := args[1]
+	var archStr string
+	if len(args) > 2 {
+		archStr = args[2]
+	} else {
+		archStr = arch.Current().String()
+	}
+
+	return outputManifest(osStdout, distroName, imgType, archStr)
+}
+
+func cmdBuild(cmd *cobra.Command, args []string) error {
+	distroName := args[0]
+	imgType := args[1]
+
+	return buildImage(osStdout, distroName, imgType)
+}
+
+func run() error {
+	// images logs a bunch of stuff to Debug/Info that we we do not
+	// want to show
+	logrus.SetLevel(logrus.WarnLevel)
+
+	rootCmd := &cobra.Command{
+		Use:   "image-builder",
+		Short: "Build operating system images from a given blueprint",
+		Long: `Build operating system images from a given blueprint
+
+Image-builder builds operating system images for a range of predefined
+operating sytsems like centos and RHEL with easy customizations support.`,
+	}
+
+	// XXX: this will list 802 images right now, we need a sensible
+	// default here, maybe without --filter just list all available
+	// distro names?
```

Comment:
I think it's fine to list everything if it's unfiltered.  Perhaps in the human-readable version we can have a count on the last line with a suggestion to look into `--filter`, so it's clear that the user is getting too much output.
For the machine readable case, listing all is definitely ok as far as I'm concerned.